Source/WTF/wtf/HashTable.h

```diff
@@ -535,6 +516,14 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(HashTable);
         static bool isDeletedBucket(const ValueType& value) { return KeyTraits::isDeletedValue(Extractor::extract(value)); }
         static bool isEmptyOrDeletedBucket(const ValueType& value) { return isEmptyBucket(value) || isDeletedBucket(value); }
 
+        template<ShouldValidateKey shouldValidateKey>
+        void validateKey(const ValueType& value)
+        {
+            if constexpr (!ASSERT_ENABLED && shouldValidateKey == ShouldValidateKey::No)
+                return;
+            RELEASE_ASSERT(!isEmptyOrDeletedBucket(value));
```

Comment:
Maybe it’s clearer written this way:
```suggestion
            if constexpr (shouldValidateKey == ShouldValidateKey::No)
                ASSERT(!isEmptyOrDeletedBucket(value));
            else
                RELEASE_ASSERT(!isEmptyOrDeletedBucket(value));
```
I think the above is what is accomplished here!?